tutorial6_hierarchical/demo/main.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * CONSTANTS AND GLOBALS
+ * */
+const width = window.innerWidth * 0.9,
+  height = window.innerHeight * 0.7,
+  margin = { top: 20, bottom: 50, left: 60, right: 40 },
+  tooltipSpecs = { width: 100, height: 50 }
```

Comment:
I kinda don't want to set the height because then we miss the benefit of using an automatically re-sizing div -- will it work without?